### PR TITLE
New NPC, Hateffects & Skill Constants

### PR DESCRIPTION
**Pull request by AoShinRO**
* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**: a lot of missing new npcs constants, some newer skills and newer hateffects

### CODE REVIEW

File: src/map/npc.hpp [hunk 1]
```diff
@@ -1536,6 +1541,1198 @@ enum e_job_types
...
+	JT_G_ILL_OBEAUNE,
+	JT_G_ILL_DEVIACE,
+	JT_G_ILL_MARSE,
+	JT_G_ILL_MERMAN,
+	JT_G_ILL_SEDORA,
+	JT_G_ILL_SWORD_FISH,
+	JT_G_ILL_STROUF,
+	JT_G_ILL_PHEN,
+	
+	JT_WANDERING_DUCK = 21089,
```
Comment (AoShinRO) [inline src/map/npc.hpp:1979]: i need to finish cleaning sequential numbers

File: src/map/npc.hpp [hunk 2]
```diff
@@ -1536,6 +1541,1212 @@ enum e_job_types
...
+	JT_ECO_ANCIENT_MUMMY,
+	JT_ECO_LUDE,
+	JT_ECO_RAGGED_ZOMBIE,
+	JT_ECO_WRAITH,
+	JT_ECO_FLAME_SKULL,
+	JT_ECO_BANSHEE,
+	JT_ECO_KNIGHT_OF_ABYSS,
+	JT_ECO_LORD_OF_DEATH,
+	JT_G_BIJOU,
+	JT_NPC_EDDGA = 21587,
```
Comment (AoShinRO) [inline src/map/npc.hpp:2316]:
```suggestion

	JT_NPC_EDDGA = 21587,
```

File: src/map/npc.hpp [hunk 3]
```diff
@@ -1536,6 +1541,1212 @@ enum e_job_types
...
+	JT_JP_E_MONSTER_146,
+	JT_JP_E_MONSTER_149,
+	JT_JP_E_MONSTER_150,
+
+	JT_EXTRA_JOKER = 20175,
+	JT_ERZSEBET,
+	JT_JENIFFER,
+	JT_GENERAL_ORC,
+	JT_SIEGLOUSE,
+	JT_VH_AMON_RA = 20181,
```
Comment (AoShinRO) [inline src/map/npc.hpp:1579]:
```suggestion

	JT_VH_AMON_RA = 20181,
```

File: src/map/npc.hpp [hunk 4]
```diff
@@ -1536,6 +1541,1214 @@ enum e_job_types
...
+	JT_MD_SAKRAY,
+	JT_MD_TIARA,
+	JT_MD_UNDEAD_KNIGHT,
+	JT_MD_UNDEAD_SOLDIER,
+	JT_MD_UNDEAD_ARCHER,
+	JT_MD_UNDEAD_WIZARD,
+	JT_MD_UNDEAD_MAGICIAN,
+	JT_MD_UNDEAD_NOBLE,
+	JT_MD_UNDEAD_SERVANT,
+	JT_CHIMERA_LAVA = 20920,
```
Comment (AoShinRO) [inline src/map/npc.hpp:2016]:
```suggestion

	JT_CHIMERA_LAVA = 20920,
```

File: src/map/npc.hpp [hunk 5]
```diff
@@ -1536,6 +1541,1214 @@ enum e_job_types
...
+	JT_GAN_CEANN,
+	JT_DISGUISER,
+	JT_BRUTAL_MURDERER,
+	JT_GHOST_CUBE,
+	JT_LUDE_GAL,
+	JT_BLUEMOON_LOLI_RURI,
+	JT_GROTE,
+	JT_PIERROTZOIST,
+	JT_DEATH_WITCH,
+	JT_MD_BETELGEUSE = 20994,
```
Comment (AoShinRO) [inline src/map/npc.hpp:2040]:
```suggestion

	JT_MD_BETELGEUSE = 20994,
```

File: src/map/npc.hpp [hunk 6]
```diff
@@ -1536,6 +1541,1214 @@ enum e_job_types
...
+	JT_BLUEMOON_LOLI_RURI2,
+	JT_FARMILIAR2,
+	JT_VR_AGNES,
+	JT_20TH_GATE_GREEN,
+	JT_20TH_GATE_PURPLE,
+	JT_20TH_GATE_RED,
+	JT_20TH_CRYSTAL_S,
+	JT_20TH_CRYSTAL_M,
+	JT_20TH_CRYSTAL_L,
+	JT_CWHITEKNIGHT = 21650,
```
Comment (AoShinRO) [inline src/map/npc.hpp:2351]:
```suggestion

	JT_CWHITEKNIGHT = 21650,
```

**Comment by AoShinRO** [comment]
> IDs in `NPC_RANGE4_*` are considered monsters, see mob_db.yml

i  can change NPC_3RD_END?

**Comment by AoShinRO** [comment]
> IDs in `NPC_RANGE4_*` are considered monsters, see mob_db.yml

can you explain better?  i did'nt find any monster with 20000 id in mob_db

**Comment by Atemo** [comment]
All IDs above 20000 are reserved for monster, others are for NPCs

**Comment by AoShinRO** [comment]
> All IDs above 20000 are reserved for monster, others are for NPCs

but is ids in here is extracted from npcidentity.lub :O

**Comment by Atemo** [comment]
Why close it? It looked good

**Comment by Haydrich** [comment]
> Why close it? It looked good

He closed all of his PR. I guess he is annoyed.

**Comment by AoShinRO** [comment]
> > Why close it? It looked good
> 
> He closed all of his PRs. I guess he is annoyed.

I'm not annoyed, just disappointed, but this is not the place to talk about that.

**Comment by AoShinRO** [comment]
> IDs in `NPC_RANGE4_*` are considered monsters, see mob_db.yml

npcs upper to 20k ids can be found at https://nn.ai4rei.net/dev/npclist/
image

**Comment by Atemo** [comment]
These NPCs have a walking animation from what I know. These special type are considered monsters in our system to categorize them as NPCs that can walk. See https://github.com/rathena/rathena/blob/master/src/map/clif.cpp#L371
Unless they changed the logic, NPCs < 20000 ID are basic NPCs, others are considered mobs

**Comment by AoShinRO** [comment]
> These NPCs have a walking animation from what I know. These special type are considered monsters in our system to categorize them as NPCs that can walk. See https://github.com/rathena/rathena/blob/master/src/map/clif.cpp#L371 Unless they changed the logic, NPCs > 20000 are considered mobs on our end

so it's like a "hack" by gravity to make NPCs walk without having to code it for NPCs, interesting haha

**Comment by Haydrich** [comment]
> > > Why close it? It looked good
> > 
> > 
> > He closed all of his PRs. I guess he is annoyed.
> 
> I'm not annoyed, just disappointed, but this is not the place to talk about that.

Hi Sir. Can i have your discord? i want to DM you. Thanks!

**Comment by AoShinRO** [comment]
> > > > Why close it? It looked good
> > > 
> > > 
> > > He closed all of his PRs. I guess he is annoyed.
> > 
> > 
> > I'm not annoyed, just disappointed, but this is not the place to talk about that.
> 
> Hi Sir. Can i have your discord? i want to DM you. Thanks!

ID: aoshinho
oldid: AoShinHo#7544

**Pull request closed without merging**